src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/LatencySensitiveLoadBalancingPolicy.java

```diff
@@ -0,0 +1,318 @@
+import com.datastax.oss.driver.api.core.session.Request;
+import com.datastax.oss.driver.api.core.session.Session;
+import com.datastax.oss.driver.internal.core.loadbalancing.helper.MandatoryLocalDcHelper;
+import com.datastax.oss.driver.internal.core.util.ArrayUtils;
+import com.datastax.oss.driver.internal.core.util.collection.QueryPlan;
+import com.datastax.oss.driver.internal.core.util.collection.SimpleQueryPlan;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import net.jcip.annotations.ThreadSafe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The default load balancing policy implementation.
+ *
+ * <p>To activate this policy, modify the {@code basic.load-balancing-policy} section in the driver
+ * configuration, for example:
+ *
+ * <pre>
+ * datastax-java-driver {
+ *   basic.load-balancing-policy {
+ *     class = DefaultLoadBalancingPolicy
+ *     local-datacenter = datacenter1
+ *   }
+ * }
+ * </pre>
+ *
+ * <p>See {@code reference.conf} (in the manual or core driver JAR) for more details.
+ *
+ * <p><b>Local datacenter</b>: This implementation requires a local datacenter to be defined,
+ * otherwise it will throw an {@link IllegalStateException}. A local datacenter can be supplied
+ * either:
+ *
+ * <ol>
+ *   <li>Programmatically with {@link
+ *       com.datastax.oss.driver.api.core.session.SessionBuilder#withLocalDatacenter(String)
+ *       SessionBuilder#withLocalDatacenter(String)};
+ *   <li>Through configuration, by defining the option {@link
+ *       DefaultDriverOption#LOAD_BALANCING_LOCAL_DATACENTER
+ *       basic.load-balancing-policy.local-datacenter};
+ *   <li>Or implicitly, if and only if no explicit contact points were provided: in this case this
+ *       implementation will infer the local datacenter from the implicit contact point (localhost).
+ * </ol>
+ *
+ * <p><b>Query plan</b>: This implementation prioritizes replica nodes over non-replica ones; if
+ * more than one replica is available, the replicas will be shuffled; if more than 2 replicas are
+ * available, they will be ordered from most healthy to least healthy ("Power of 2 choices" or busy
+ * node avoidance algorithm). Non-replica nodes will be included in a round-robin fashion. If the
+ * local datacenter is defined (see above), query plans will only include local nodes, never remote
+ * ones; if it is unspecified however, query plans may contain nodes from different datacenters.
+ */
+@ThreadSafe
+public class LatencySensitiveLoadBalancingPolicy extends DefaultLoadBalancingPolicy {
```

Comment:
It looks like the class-level Javadoc here was lifted from [DefaultLoadBalancingPolicy](https://github.com/apache/cassandra-java-driver/blob/4.18.1/core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/DefaultLoadBalancingPolicy.java).  It's probably more useful to describe this as a subclass of DefaultLoadBalancingPolicy which behaves differently in a few cases... and then explain how it behaves differently.